packages/relay-compiler/index.d.ts

```diff
@@ -0,0 +1,327 @@
+   * (TypeScript only) Whether to use the `import type` syntax introduced in Typescript version 3.8. This will prevent warnings from `importsNotUsedAsValues`.
+   */
+  useImportTypeSyntax?: boolean;
+  customScalarTypes?: CustomScalarTypes;
+  /**
+   * Require all GraphQL scalar types mapping to be defined, will throw if a GraphQL scalar type doesn't have a JS type.
+   */
+  requireCustomScalarTypes?: boolean;
+  /**
+   * Work in progress new Flow type definitions
+   */
+  flowTypegen?: {
+    /**
+     * This option controls whether or not a catch-all entry is added to enum type definitions values that may be added 
+     * in the future. Enabling this means you will have to update your application whenever the GraphQL server schema adds 
+     * new enum values to prevent it from breaking.
+     * 
+     * @default false
+     */
+    noFutureProofEnums?: NoFutureProofEnums;
+  };
+  /**
+   * This option enables emitting ES modules artifacts.
+   * 
+   * @default false
+   */
+  eagerEsModules?: boolean;
+}
+
+export type RelayConfigMultiProjectItem = 
+  RelayConfigMultiProjectItemSchemaConfig & RelayConfigMultiProjectItemTypegenConfig & {
+  /**
+   * The name of the base project. If a base project is set, the documents of that project can be referenced, 
+   * but won't produce output artifacts. Extensions from the base project will be added as well and the schema 
+   * of the base project should be a subset of the schema of this project.
+   */
+  base?: string;
+  /**
+   * The output directory for all our generated artifactories from the compiler. A project without an output directory will put 
+   * the generated files in a `__generated__` directory next to the input file. All files in these directories should be generated 
+   * by the Relay compiler, so that the compiler can cleanup extra files.
+   */
+  output?: string;
+  /**
+   * Some projects may need to generate extra artifacts. For those, we may need to provide an additional directory to put them. 
+   * By default they will use `output`.
+   */
+  extraArtifactsOutput?: string;
+  /**
+   * If `output` is provided and `shard_output` is `true`, shard the files by putting them under `{output_dir}/{source_relative_path}`
+   */
+  shardOutput?: boolean;
+  /**
+   * Regex to match and strip parts of the `source_relative_path`
+   */
+  shardStripRegex?: string;
+  /** Array of directories containing *.graphql files with schema extensions. */
+  schemaExtensions?: string[];
+  /** If this option is set, the compiler will persist queries using this config. */
+  persist?: RelayConfigPersitConfigLocal;
+  /**
+   * Restrict @relay_test_operation to directories matching this regex
+   */
+  testPathRegex?: string;
+  /** Generates a `// @relayVariables name1 name2` header in generated operation files. */
+  variableNamesComment?: boolean;
+  featureFlags?: RelayConfigFeatureFlags;
+  /** A placeholder for allowing extra information in the config file */
+  extra: unknown;
```

Comment:
Presumably this should be optional?

Reply:
This should probably just be `[configKey: string]: any`?

Reply:
I pushed a fix into this tht turns it into an optional field.

> This should probably just be `[configKey: string]: any`?

In case, I think that if we should type it, we should type accordingly to the `serde::Value` struct, that can be any JSON-friendly value. What do you think?